gnmi/reconciler/reconciler.go

```diff
@@ -41,3 +45,124 @@ type Reconciler interface {
 	// ValidationPaths returns the set of path prefixes that the reconciler can validate.
 	ValidationPaths() []ygnmi.PathStruct
 }
+
+// Builder simplifies the creation of reconcilers and reduces some of the required boilerplate.
+type Builder struct {
+	br *builtReconciler
+}
+
+// Build returns the reconciler as configuration and resets the builder.
+func (b *Builder) Build() Reconciler {
+	reconciler := b.br
+	b.br = &builtReconciler{}
+	return reconciler
+}
```

Comment:
sure BuiltReconciler can be exported. i put the reconciler interface here for discoverability

Reply:
SG

Reply:
The return should also be BuiltReconciler -- the user can use the interface definition if they wish. If we later add more methods to BuiltReconciler then this will be exposed right away instead of having to refactor, e.g. if we added another reconciler implementation that fixed the definition of `Reconciler`.